spdx/parsers/rdf.py

```diff
@@ -561,10 +561,11 @@ def p_pkg_files_analyzed(self, p_term, predicate):
     def p_pkg_originator(self, p_term, predicate):
         for _s, _p, o in self.graph.triples((p_term, predicate, None)):
             try:
-                if o == "NOASSERTION":
-                    self.builder.set_pkg_originator(self.doc, utils.NoAssert())
+                originator = self.to_special_value(o)
+                if isinstance(originator, utils.NoAssert):
+                    self.builder.set_pkg_originator(self.doc, originator)
                 else:
-                    ent = self.builder.create_entity(self.doc, str(o))
+                    ent = self.builder.create_entity(self.doc, originator)
```

Comment:
At this point, `originator` could be `utils.SPDXNone`, but `create_entity` expects it to be a string.

See below for `supplier`, too.

Reply:
True, it could also be `utils.UnKnown`. This isn't explicitly allowed or forbidden by the spec, so I simply added these to the check for instances above. As we are not fully supporting rdf here, I think this is alright.